### DISCUSSION TITLE
Generated code creates conflicts whenever a new scheme is used on a new run

### DISCUSSION THREAD

**Discussion by erdtsieck** [JasperFx/marten#2733] [General]
Whenever you setup a marten integration test like this:

```cs
public string SchemaName { get; } = "sch" + Guid.NewGuid().ToString().Replace("-", string.Empty);

[Fact]
public async Task Isolated()
{
    var host = await AlbaHost.For<Program>(b =>
    {
        b.ConfigureTestServices(services =>
        {
            services.AddMarten(options =>
            {
                options.Connection(ConnectionString);
                options.Events.DatabaseSchemaName = SchemaName;
                options.DatabaseSchemaName = SchemaName;
            });
        });
    });

    // Arrange
    await using var session = host.Services.GetRequiredService<IDocumentSession>();
    var tijdlijnitem = new Fixture().Build<TijdlijnItem>().Create();
    session.Store(tijdlijnitem);
    await session.SaveChangesAsync();

    // Act
    var all = session.Query<TijdlijnItem>().ToList();
    
    // Assert
    Assert.NotEmpty(all);
}
```

And you're using `.OptimizeArtifactWorkflow()` code is generated just fine in the first run, but all additional runs no new code is generated. This is problematic since the generated code contains the scheme name. 

It's extra problematic as the code is generated in the actual application and not in the integration test projects. So you could by accident check-in wrong generated code.

**Comment by jeremydmiller** [comment]
@erdtsieck This is going to be a documentation issue if anything. You just can't do both things at the same time is the only possible answer here. 

I'm converting this to a discussion. We can add something to the docs I guess.

**Reply by erdtsieck** [reply]
I added a workaround to the integration testing docs: https://github.com/JasperFx/marten/pull/2732
